chesster.js

```diff
@@ -1365,23 +1366,48 @@ controller.on('ambient', function(bot, message) {
                 return;
             }
 
-            spreadsheets.update_result(
+            //this could and probably should be improved at some point
+            //this will require two requests to the spread sheet and
+            //it can be done in one, but I am trying to reuse what 
+            //I wrote before as simply as possibe for now
+
+            //if a gamelink already exists, get it
+            spreadsheets.fetch_pairing_gamelink(
                 config.service_account_auth,
                 results_options.key,
                 results_options.colname,
-                result, 
-                function(err, reversed){
-                    if (err) {
-                        if (err.indexOf && err.indexOf("Unable to find pairing.") == 0) {
-                            result_reply_missing_pairing(bot, message);
-                        } else {
-                            bot.reply(message, "Something went wrong. Notify @endrawes0");
-                            throw new Error("Error updating scheduling sheet: " + err);
-                        }
-                    } else {
-                        result_reply_updated(bot, message, result);
+                result,
+                function(err, gamelink){
+                    //if a gamelink is found, use it to acquire details and process them
+                    if(!err && gamelink){
+                        process_gamelink(
```

Comment:
OK - I have implemented this. I agree it is important. I will need to do more testing to make sure it is working correctly.